tf2_ros/src/transform_listener.cpp

```diff
@@ -67,13 +67,64 @@ TransformListener::~TransformListener()
 }
 
 
+inline
+bool
+_make_authority_str(
+  const std::vector<rclcpp::TopicEndpointInfo> & endpoints,
+  const rmw_gid_t & pub_gid,
+  std::string & authority)
+{
+  for (const rclcpp::TopicEndpointInfo & pub : endpoints) {
+    // Compare gid to see if this is the right publisher
+    bool match = true;
+    for (decltype(RMW_GID_STORAGE_SIZE) i = 0; i < RMW_GID_STORAGE_SIZE; ++i) {
```

Comment:
I'm somewhat concerned about the performance impact of this check, particularly on networks with lots of `/tf` publishers.  While I know that the inner loop has a fixed number of iterations, it is still the case that we are going to be doing this linear search on every single reception of a `tf2` frame, which seems expensive.  Should we consider a `map` or an `unordered_map` here?